persistence-jpa/javadsl/src/main/java/com/lightbend/lagom/internal/javadsl/persistence/jpa/JpaReadSideImpl.java

```diff
@@ -149,7 +149,11 @@ private JpaReadSideHandler(
                     if (log.isDebugEnabled())
                         log.debug("Unhandled event {} at offset {} in JpaReadSideHandler: {}",
                                 eventClass.getName(), offset, readSideId);
-                    return CompletableFuture.completedFuture(Done.getInstance());
+                    return jpa.withTransaction(entityManager -> {
+
+                        updateOffset(entityManager, offset);
+                        return Done.getInstance();
+                    });
```

Comment:
We can't do much here. JPA won't allow us to easily compose actions. Only option is to repeat stuff